sympy/matrices/expressions/applyfunc.py

```diff
@@ -72,19 +72,28 @@ def func(self):
         # This strange construction is required by the assumptions:
         # (.func needs to be a class)
 
-        class ElementwiseApplyFunction2(ElementwiseApplyFunction):
-            def __new__(obj, expr):
+        class _(ElementwiseApplyFunction):
+            def __new__(cls, expr):
                 return ElementwiseApplyFunction(self.function, expr)
 
-        return ElementwiseApplyFunction2
+        return _
 
     def doit(self, **kwargs):
         deep = kwargs.get("deep", True)
         expr = self.expr
         if deep:
             expr = expr.doit(**kwargs)
+        function = self.function
+        if isinstance(function, Lambda) and function.args[0] == (function.args[1],):
+            # This is a Lambda containing the identity function.
+            return expr
```

Comment:
Lambda has `.is_identity` for this. Returning `function(expr)` will check that the signature of the function is actually compatible though e.g.:
```julia
In [1]: function = Lambda((x, y), x+y)                                                                                            

In [2]: expr = 2                                                                                                                  

In [3]: function(2)                                                                                                               
---------------------------------------------------------------------------
BadArgumentsError
```

Reply:
I guess a better example would be:
```julia
In [7]: function = Lambda((x, y), (x, y))                                                                                         

In [8]: function.is_identity                                                                                                      
Out[8]: True
```